sklearn/cross_validation.py

```diff
@@ -700,41 +706,90 @@ def __len__(self):
 
 def _validate_shuffle_split(n, test_size, train_size):
     if test_size is None and train_size is None:
-        raise ValueError(
-            'test_size and train_size can not both be None')
+        # user didn't set test_size
+        # we will use default value
+        test_size = 0.1
+        train_size = 1. - test_size
 
-    if test_size is not None:
+    if test_size is not None and train_size is None:
         if np.asarray(test_size).dtype.kind == 'f':
             if test_size >= 1.:
                 raise ValueError(
                     'test_size=%f should be smaller '
                     'than 1.0 or be an integer' % test_size)
+            train_size = 1. - test_size
         elif np.asarray(test_size).dtype.kind == 'i':
             if test_size >= n:
                 raise ValueError(
                     'test_size=%d should be smaller '
                     'than the number of samples %d' % (test_size, n))
+            train_size = n - test_size
         else:
             raise ValueError("Invalid value for test_size: %r" % test_size)
-
-    if train_size is not None:
+    elif test_size is None and train_size is not None:
         if np.asarray(train_size).dtype.kind == 'f':
             if train_size >= 1.:
                 raise ValueError("train_size=%f should be smaller "
                                  "than 1.0 or be an integer" % train_size)
-            elif np.asarray(test_size).dtype.kind == 'f' and \
-                    train_size + test_size > 1.:
+            test_size = 1. - train_size
+        elif np.asarray(train_size).dtype.kind == 'i':
+            if train_size >= n:
+                raise ValueError("train_size=%d should be smaller "
+                                 "than the number of samples %d" %
+                                 (train_size, n))
+            test_size = n - train_size
+        else:
+            raise ValueError("Invalid value for train_size: %r" % train_size)
+    else:  # test_size is not None and train_set is not None:
+        if np.array(test_size).dtype.kind == 'f' and \
+           np.array(train_size).dtype.kind == 'f':
+            if test_size >= 1.:
+                raise ValueError('test_size=%f should be smaller '
+                                 'than 1.0 or be an integer' % test_size)
+            if train_size >= 1.:
+                raise ValueError('train_size=%f should be smaller '
+                                 'than 1.0 or be an integer' % train_size)
+            if train_size + test_size > 1.:
                 raise ValueError('The sum of test_size and train_size = %f, '
                                  'should be smaller than 1.0. Reduce '
                                  'test_size and/or train_size.' %
                                  (train_size + test_size))
-        elif np.asarray(train_size).dtype.kind == 'i':
+        elif np.array(test_size).dtype.kind == 'f' and \
```

Comment:
Also maybe just check if ``np.asarray(train_size).dtype.kind == np.asarray(test_size).dtype.kind``, not two explicit tests.